src/js/components/validateData/validateValues/ValidateValuesFileComponent.jsx

```diff
@@ -84,6 +86,22 @@ export default class ValidateDataFileComponent extends React.Component {
         return stagedFile;
     }
 
+    getFileUrl(item){
+        if(item.file_type=='appropriations'){
+            GenerateFilesHelper.fetchFile('A', this.props.submission.id).then((result)=>{
+                this.setState({url: result.url});
+            })
+        }else if(item.file_type=='program_activity'){
+            GenerateFilesHelper.fetchFile('B', this.props.submission.id).then((result)=>{
+                this.setState({url: result.url});
+            })
+        }else if(item.file_type=='award_financial'){
+            GenerateFilesHelper.fetchFile('C', this.props.submission.id).then((result)=>{
+                this.setState({url: result.url});
+            })
+        }
+    }
+
```

Comment:
The problem with this method of setting the url is that the url can expire if the user sits on this page for too long and they have to refresh to be able to access it again. Take a look at `components/validateData/validateValues/ValidateValuesErrorReport.jsx` for a sample of how to get the URL on click.

I know we use this method in other places as well, but not all of them have been updated to generate the URL on click so some URLs can still expire